internal/gatewayapi/translator_test.go

```diff
@@ -76,8 +78,8 @@ func TestTranslate(t *testing.T) {
 			got := translator.Translate(resources)
 
 			sort.Slice(got.XdsIR.HTTP, func(i, j int) bool { return got.XdsIR.HTTP[i].Name < got.XdsIR.HTTP[j].Name })
-
-			assert.EqualValues(t, want, got)
+			opts := cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime")
```

Comment:
Instead of ignoring this field, should we have a test case that should cause a condition to __not__ be updated and ensure the `LastTransitionTime` is not updated?

Reply:
this needs to be ignored for all the YAML based test cases we have
will add a separate test case around LastTransitionTime

Reply:
based on the current logic `LastTransitionTime` will change 
https://github.com/envoyproxy/gateway/blob/3af2ddf27b74222cd525347942a8df9201dd79a6/internal/gatewayapi/contexts.go#L37
if you feel this is a bug an issue should be raised

Reply:
`mergeConditions()` should only update status if the condition `Status`, `Reason` or `Message` has changed and set the `lastTransitionTime` accordingly.

I reproduced the expected behavior by starting EG, and creating a GatewayClass. Here are the EG logs:

```
2022-09-20T17:53:38.100Z  INFO  eg  kubernetes/gatewayclass.go:83 enqueueing gatewayclass {"runner": "provider"}
2022-09-20T17:53:38.100Z  INFO  eg  kubernetes/gatewayclass.go:92 reconciling gatewayclass  {"runner": "provider"}
2022-09-20T17:53:38.101Z  INFO  eg  kubernetes/gatewayclass.go:169  reconciled gatewayclass {"runner": "provider"}
2022-09-20T17:53:38.101Z  INFO  status/status.go:103  received a status update  {"runner": "provider", "namespace": "", "name": "eg"}
```

and the initial gatewayclass status:
```
$ kubectl get gc -o yaml
...
  status:
    conditions:
    - lastTransitionTime: "2022-09-20T17:53:38Z"
      message: Valid GatewayClass
      observedGeneration: 1
      reason: Accepted
      status: "True"
      type: Accepted
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

I then create a gateway which triggers a gatewayclass reconciliation:
```
2022-09-20T17:54:54.386Z  INFO  kubernetes/gateway.go:140 reconciling gateway {"runner": "provider", "namespace": "default", "name": "eg"}
2022-09-20T17:54:54.415Z  INFO  default.eg  kubernetes/gateway.go:226 reconciled gateway  {"runner": "provider"}
2022-09-20T17:54:54.415Z  INFO  status/status.go:103  received a status update  {"runner": "provider", "namespace": "default", "name": "eg"}
2022-09-20T17:54:54.415Z  INFO  runner/runner.go:68 received a notification {"runner": "gateway-api"}
2022-09-20T17:54:54.415Z  INFO  eg  kubernetes/gatewayclass.go:83 enqueueing gatewayclass {"runner": "provider"}
2022-09-20T17:54:54.416Z  INFO  eg  kubernetes/gatewayclass.go:92 reconciling gatewayclass  {"runner": "provider"}
2022-09-20T17:54:54.416Z  INFO  eg  kubernetes/gatewayclass.go:169  reconciled gatewayclass {"runner": "provider"}
...
2022-09-20T17:54:54.453Z  INFO  status/status.go:103  received a status update  {"runner": "provider", "namespace": "", "name": "eg"}
2022-09-20T17:54:54.453Z  INFO  eg  status/status.go:75 status unchanged, bypassing update  {"runner": "provider"}
```

The `lastTransitionTime` of the gatewayclass is not updated:
```
$ kubectl get gc -o yaml
...
  status:
    conditions:
    - lastTransitionTime: "2022-09-20T17:53:38Z"
      message: Valid GatewayClass
      observedGeneration: 1
      reason: Accepted
      status: "True"
      type: Accepted
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

This is the expected behavior. I also tested by restarting EG and `lastTransitionTime` was not updated.

@arkodg I don't see a bug in the current code, can you please elaborate?

Reply:
the above code points to `Gateway` status and not `GatewayClass` . currently `mergeConditions` will update the entire condition in that index (including lastTransitionTime) if the condition `Type` matches https://github.com/envoyproxy/gateway/blob/3af2ddf27b74222cd525347942a8df9201dd79a6/internal/gatewayapi/contexts.go#L28

Reply:
OK, I see now. I find it confusing that the gatewayclass/gateway controllers compute and update status conditions, e.g. `status.Updater` for GatewayClasses and Gateways while the gatewayapi translator computes ListenerStatus. @arkodg @skriss @youngnick @LukeShu thoughts on consolidating on one approach to managing status?